adapter/src/hermes/adapter/stdio/stdio.cc

```diff
@@ -0,0 +1,1017 @@
+                    << " blobs to filename:" << filename << "." << std::endl;
+          INTERCEPTOR_LIST->hermes_flush_exclusion.insert(filename);
+          hermes::api::VBucket file_vbucket(filename, mdm->GetHermes(), true,
+                                            ctx);
+          auto offset_map = std::unordered_map<std::string, hermes::u64>();
+          for (const auto &blob_name : blob_names) {
+            file_vbucket.Link(blob_name, filename, ctx);
+            offset_map.emplace(blob_name, std::stol(blob_name) * kPageSize);
+          }
+          auto trait = hermes::api::FileMappingTrait(filename, offset_map,
+                                                     nullptr, NULL, NULL);
+          file_vbucket.Attach(&trait, ctx);
+          file_vbucket.Delete(ctx);
+          existing.first.st_blobs.clear();
+          INTERCEPTOR_LIST->hermes_flush_exclusion.erase(filename);
+        }
+        existing.first.st_bkid->Destroy(ctx);
+        mdm->FinalizeHermes();
+      } else {
+        LOG(INFO) << "File handler is opened by more than one fopen."
+                  << std::endl;
+        existing.first.ref_count--;
+        struct timespec ts;
+        timespec_get(&ts, TIME_UTC);
+        existing.first.st_atim = ts;
+        existing.first.st_ctim = ts;
+        mdm->Update(fp, existing.first);
+        existing.first.st_bkid->Close(ctx);
+      }
+    }
+  }
+
+  MAP_OR_FAIL(fclose);
+  ret = real_fclose_(fp);
+  return (ret);
+}
+
+size_t HERMES_DECL(fwrite)(const void *ptr, size_t size, size_t nmemb,
+                           FILE *fp) {
+  size_t ret;
+  if (hermes::adapter::IsTracked(fp)) {
+    auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
+    auto existing = mdm->Find(fp);
+    if (existing.second) {
+      LOG(INFO) << "Intercept fwrite." << std::endl;
+      ret = write_internal(existing, ptr, size * nmemb, fp);
+    } else {
+      MAP_OR_FAIL(fwrite);
+      ret = real_fwrite_(ptr, size, nmemb, fp);
+    }
+  } else {
+    MAP_OR_FAIL(fwrite);
+    ret = real_fwrite_(ptr, size, nmemb, fp);
+  }
+  return (ret);
+}
+
+int HERMES_DECL(fputc)(int c, FILE *fp) {
+  int ret;
+  if (hermes::adapter::IsTracked(fp)) {
+    auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
+    auto existing = mdm->Find(fp);
+    if (existing.second) {
+      LOG(INFO) << "Intercept fputc." << std::endl;
+      write_internal(existing, &c, 1, fp);
+      ret = c;
+    } else {
+      MAP_OR_FAIL(fputc);
+      ret = real_fputc_(c, fp);
+    }
+  } else {
+    MAP_OR_FAIL(fputc);
+    ret = real_fputc_(c, fp);
+  }
+  return (ret);
+}
+
+int HERMES_DECL(fgetpos)(FILE *fp, fpos_t *pos) {
+  int ret;
+  if (hermes::adapter::IsTracked(fp) && pos) {
+    auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
+    auto existing = mdm->Find(fp);
+    if (existing.second) {
+      LOG(INFO) << "Intercept fgetpos." << std::endl;
+      pos->__pos = existing.first.st_ptr;
```

Comment:
Could you please add the following comment in the 3 functions (`fgetpos`, `fsetpos`, and `fsetpos64`) where you access `fpos_t->__pos`:

```c++
// TODO(chogan): @portability In the GNU C Library, fpos_t is an opaque data
// structure that contains internal data to represent file offset and conversion
// state information. In other systems, it might have a different internal
// representation. This will need to change to support other compilers.

```

Reply:
Added